core/integration/storagetest/mutation_logs_reader.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagetest
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/keytransparency/core/sequencer"
+
+	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
+)
+
+// LogsReadWriter supports test's ability to write to and read from the mutation logs.
+type LogsReadWriter interface {
+	sequencer.LogsReader
+	// Send submits the whole group of mutations atomically to a log.
+	// TODO(gbelvin): Create a batch level object to make it clear that this a batch of updates.
+	// Returns the timestamp that the mutation batch got written at.
```

Comment:
I think a note on this returned value about whether it can be used directly in a query, or whether it needs 1 adding to it will help put a stake in the ground about this topic.